contracts/extensions/MultisigPermissions.sol

```diff
@@ -0,0 +1,640 @@
+      // We set the overall approval timestamp to now, and return
+      // We don't execute the motion, but we want to commit the timestamp
+      //  (which wouldn't happen if we continued and a call failed)
+      motion.overallApprovalTimestamp = block.timestamp;
+      return;
+    }
+
+    motion.executed = true;
+    bool overallSuccess = true;
+
+    for (uint256 i = 0; i < motion.data.length; i += 1) {
+      // solhint-disable-next-line avoid-low-level-calls
+      (bool success, ) = address(motion.targets[i]).call(motion.data[i]);
+      overallSuccess = overallSuccess && success;
+
+      // Allow failing execution after seven days
+      require(
+        success || motion.overallApprovalTimestamp + 7 days <= block.timestamp,
+        "colony-multisig-failed-not-one-week"
+      );
+    }
+
+    emit MotionExecuted(msgSender(), _motionId, overallSuccess);
+  }
+
+  function checkThreshold(uint256 _motionId, Vote _vote) private view returns (bool thresholdMet) {
+    Motion storage motion = motions[_motionId];
+    uint8 roleIndex;
+
+    // While there are still relevant roles we've not checked yet
+    while (uint256(motion.requiredPermissions) >= (1 << roleIndex)) {
+      // For the current role, is it required for the motion?
+      if ((motion.requiredPermissions & bytes32(1 << roleIndex)) > 0) {
+        uint256 threshold = getDomainSkillRoleThreshold(motion.domainSkillId, roleIndex);
+        if (motionVoteCount[_motionId][_vote][roleIndex] < threshold) {
+          return false;
+        }
+      }
+      roleIndex += 1;
+    }
+
+    return true;
+  }
+
+  function getGlobalThreshold() public view returns (uint256) {
+    return globalThreshold;
+  }
+
+  function getMotionCount() public view returns (uint256) {
+    return motionCount;
+  }
+
+  function getMotion(uint256 motionId) public view returns (Motion memory) {
+    return motions[motionId];
+  }
+
+  function getMotionRoleVoteCount(
+    uint256 _motionId,
+    uint8 _role,
+    Vote _vote
+  ) public view returns (uint256) {
+    return motionVoteCount[_motionId][_vote][_role];
+  }
+
+  function getUserVote(
+    uint256 _motionId,
+    address _user,
+    uint8 _role,
+    Vote _vote
+  ) public view returns (bool) {
+    bytes32 userVotes = motionVotes[_motionId][_user][_vote];
+    return (userVotes >> uint8(_role)) & bytes32(uint256(1)) == bytes32(uint256(1));
+  }
+
+  function getDomainSkillRoleCounts(
+    uint256 _domainSkillId,
+    uint8 _role
+  ) public view returns (uint256) {
+    return domainSkillRoleCounts[_domainSkillId][_role];
+  }
+
+  function getDomainSkillRoleThreshold(
+    uint256 _domainSkillId,
+    uint8 _role
+  ) public view returns (uint256) {
+    if (domainSkillThreshold[_domainSkillId] > 0) {
+      return domainSkillThreshold[_domainSkillId];
+    }
+
+    if (globalThreshold > 0) {
+      return globalThreshold;
+    }
+
+    return (domainSkillRoleCounts[_domainSkillId][_role] / 2) + 1;
+  }
+
+  function getActionSummary(
```

Comment:
Can we adapt the implementation to use the shared GetActionSummary mixin? Seems wasteful to have a slightly different reimplementation here.

Reply:
I've done what I can the generalise this. I've ended up making the mixin `getSingleActionSummary`, as ReputationVoting and multisig permissions have different requirements, and therefore different implementations past that point:

* Reputation Voting treats the `NO_ACTION`, `OLD_MOVE_FUNDS` signatures differently
* So long as the same domain / expenditure is being targeted, any combination of required permissions is allowed. For Multisig, functions must all require the same permissions (though I'm considering relaxing this requirement - I suspect that it's actually too restrictive for what product would like)

Certainly not sold on this division of functionality, so I would certainly consider it up for further discussion, but I think it's the best we can do with these sets of requirements.

Reply:
By 'previous version', do you mean removing the mixin entirely, as it would only be used in one place?

I think there are benefits to this refactorisation though - specifically having a place where edge cases like `setUserRole` can live. What is it that you don't like about the reduced functionality in the mixin?